flutter_local_notifications/ios/Classes/FlutterEngineManager.m

```diff
@@ -0,0 +1,92 @@
+//
+//  FlutterEngineManager.h
+//  flutter_local_notifications
+//
+//  Created by Sebastian Roth on 12/30/21.
+//
+
+#import <Foundation/Foundation.h>
+
+#import <Flutter/Flutter.h>
+
+#import "ActionEventSink.h"
+#import "FlutterEngineManager.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+static FlutterEngine *backgroundEngine;
+
+@implementation FlutterEngineManager {
+  NSUserDefaults *_persistentState;
+}
+
++ (BOOL)shouldAddAppDelegateToRegistrar:
```

Comment:
Hi @ened, I know it's been a long time now since this PR but do you remember the reason for this method? I was looking to see if this could be removed. A while back, the plugin was bumped that it no longer supports old notification APIs that in turn required hooking into the app delegate. I removed the methods tied to the delegate but realised it's still registered. Looking to see if the registration can be removed. I believe so but wanted to check with you in case I've missed something

Reply:
Ah sorry I spoke too soon. I missed that removing the registration of the plugin as the app delegate had an effect on the `UNUserNotificationCenterDelegate`  methods from being fired. I forgot tjay the Flutter engine is detecting if a plugin has registered as an application delegate to then try to fire the appropriate `UNUserNotificationCenterDelegate` methods